### DISCUSSION TITLE
Views for Security (for columns)

### DISCUSSION THREAD

**Discussion by ozum** [PostgREST/postgrest#2617] [Q&A] [answered]
Hi,

I want to limit accessing columns based on the user level. For example normal users can access `a.name`, `a.surname`, but admin users can see also `a.mail`.

In Postgrest context, it seems possible to use views for the purpose. I can create a view `b` with limited fields for normal users. However I don't want to have two API endpoints `a` and `b`. Otherwise, each user level needs to use a different API endpoint for the same table.

How can I use a single table name/endpoint? (i.e. `c` for normal and  admin users). I examined PostgreSQL select rules for redirecting queries, but it seems it doesn't support my use case. Any suggestion?

**Comment by steve-chavez** [comment, marked as answer]
Hello,

> However I don't want to have two API endpoints a and b

How about using column-level grants for this, with two different roles. So you'll have:

```sql
CREATE ROLE normal;
CREATE ROLE admin;

GRANT SELECT(name, surname) ON a TO normal;
GRANT SELECT(name, surname, mail) ON a TO admin;
```

Then you can do both requests on a single endpoint `a`.

**Reply by ozum** [reply]
> How about using column-level grants for this, with two different roles

Thanks @steve-chavez, after your message I saw [hybrid user group roles](https://postgrest.org/en/stable/auth.html#hybrid-user-group-roles). Normally, I do that, but I'm using Supabase (which uses PostgREST), and it seems to support fixed database roles, but not hybrid user group roles. Sorry, I forgot to mention Supabase in the original message. OTOH, I'm going to ask about hybrid user group roles support in the Supabase repository.
